### ISSUE TITLE
[9.0] App installation fails on clean Odoo 9.0

### ISSUE THREAD

**Issue by satriani-vai** [odoo/odoo#8738]
I get error during the installation of the apps on a clean 9.0
I noticed that this happens after trying to install the eCommerce app, which fails with this error.
```
Odoo Server Error
Traceback (most recent call last):
  File "/usr/local/odoo_9/openerp/http.py", line 546, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/local/odoo_9/openerp/http.py", line 583, in dispatch
    result = self._call_function(**self.params)
  File "/usr/local/odoo_9/openerp/http.py", line 316, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/http.py", line 309, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/local/odoo_9/openerp/http.py", line 828, in __call__
    return self.method(*args, **kw)
  File "/usr/local/odoo_9/openerp/http.py", line 412, in response_wrap
    response = f(*args, **kw)
  File "/usr/local/odoo_9/addons/web/controllers/main.py", line 981, in call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/local/odoo_9/addons/web/controllers/main.py", line 969, in _call_kw
    return checked_call(request.db, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/local/odoo_9/addons/web/controllers/main.py", line 968, in checked_call
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/addons/base/module/module.py", line 459, in button_immediate_install
    return self._button_immediate_function(cr, uid, ids, self.button_install, context=context)
  File "/usr/local/odoo_9/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/addons/base/module/module.py", line 533, in _button_immediate_function
    registry = openerp.modules.registry.RegistryManager.new(cr.dbname, update_module=True)
  File "/usr/local/odoo_9/openerp/modules/registry.py", line 362, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/usr/local/odoo_9/openerp/modules/loading.py", line 337, in load_modules
    loaded_modules, update_module)
  File "/usr/local/odoo_9/openerp/modules/loading.py", line 237, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/usr/local/odoo_9/openerp/modules/loading.py", line 156, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/usr/local/odoo_9/openerp/modules/loading.py", line 98, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/usr/local/odoo_9/openerp/tools/convert.py", line 852, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/usr/local/odoo_9/openerp/tools/convert.py", line 939, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/usr/local/odoo_9/openerp/tools/convert.py", line 802, in parse
    self.parse(rec, mode)
  File "/usr/local/odoo_9/openerp/tools/convert.py", line 805, in parse
    self._tags[rec.tag](self.cr, rec, de, mode=mode)
  File "/usr/local/odoo_9/openerp/tools/convert.py", line 709, in _tag_record
    id = self.pool['ir.model.data']._update(cr, self.uid, rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode, context=rec_context )
  File "/usr/local/odoo_9/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/addons/base/ir/ir_model.py", line 1105, in _update
    res_id = model_obj.create(cr, uid, values, context=context)
  File "/usr/local/odoo_9/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/addons/base/ir/ir_ui_view.py", line 358, in create
    context=context)
  File "/usr/local/odoo_9/openerp/api.py", line 223, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/api.py", line 318, in old_api
    result = method(recs, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/models.py", line 4118, in create
    record = self.browse(self._create(old_vals))
  File "/usr/local/odoo_9/openerp/api.py", line 221, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/api.py", line 454, in new_api
    result = method(self._model, cr, uid, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/models.py", line 4309, in _create
    recs._validate_fields(vals)
  File "/usr/local/odoo_9/openerp/api.py", line 221, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/local/odoo_9/openerp/models.py", line 1267, in _validate_fields
    raise ValidationError('\n'.join(errors))
ParseError: "Invalid view definition

Error details:
Field `product_id` does not exist

Error context:
View `product.supplierinfo.form.view`
[view_id: 493, xml_id: n/a, model: product.supplierinfo, parent_id: n/a]
None" while parsing /usr/local/odoo_9/addons/product/product_view.xml:762, near
<record id="product_supplierinfo_form_view" model="ir.ui.view">
            <field name="name">product.supplierinfo.form.view</field>
            <field name="model">product.supplierinfo</field>
            <field name="arch" type="xml">
                <form string="Vendor Information">
                    <group>
                        <group string="Vendor">
                            <field name="name" context="{'default_customer': 0, 'search_default_supplier': 1, 'default_supplier': 1}"/>
                            <field name="product_name"/>
                            <field name="product_code"/>
                            <field name="product_id" domain="[('product_tmpl_id', '=', product_tmpl_id)]" invisible="1"/>
                            <label for="delay"/>
                            <div>
                                <field name="delay" class="oe_inline"/> days
                            </div>
                        </group>
                        <group string="Price List">
                            <field name="product_tmpl_id" string="Product" invisible="context.get('visible_product_tmpl_id', True)"/>
                            <label for="min_qty"/>
                            <div>
                                <field name="min_qty" class="oe_inline"/>
                                <field name="product_uom" class="oe_inline" groups="product.group_uom"/>
                            </div>
                            <label for="price"/>
                            <div>
                                <field name="price" class="oe_inline"/><field name="currency_id" class="oe_inline" groups="base.group_multi_currency"/>
                            </div>
                            <label for="date_start" string="Validity"/>
                            <div><field name="date_start" class="oe_inline"/> to <field name="date_end" class="oe_inline"/></div>
                        </group>
                        <group string="Other Information" groups="base.group_multi_company">
                            <field name="company_id" options="{'no_create': True}"/>
                        </group>
                    </group>
                </form>
            </field>
        </record>
```

**Comment by erokata** [comment]
Same here after Point of Sale installation.

**Comment by ddrapeau** [comment]
When I look in the __openerp__.py file, I can see there is no "product" in dict key 'depends'. And product_id need to have the module "Product" installed to work. Try to add 'product' in __openerp__.py file wich is addons/product/ and in this file you have a dictionary with a key witch is "depends".

**Comment by satriani-vai** [comment]
Thank you ddrapeau,
now we can see, this is a depends issue. However, this need to be fixed.

**Comment by ddrapeau** [comment]
I have tried to commit change and to push on repository but without success. 
@fpodoo is it possible to commit and push fixes on a branch ?

**Comment by apr-odoo** [comment]
duplicate of #8762

**Comment by mart-e** [comment]
The field `product_id ` was added at b65c67a2598fd3ad433af05ed49d398c473b76ae. So if your database is older than this commit but you just updated your code, it will probably crash.

**Comment by mart-e** [comment]
@ddrapeau the `__openerp__.py` file do not specify `product` in the `depends` section but it is indirect. `website_sale` -> `sale` -> `procurement` -> `product`.
And to submit a fix, you need to push on a repository that belongs to you and then submit a pull request.

Closing as fixed with a new database (9.0 is not stable yet) but let me know if you can reproduce with a new db and we will reopen.
